pkg/agent/protocol/protocol.go

```diff
@@ -29,9 +29,21 @@ type Disruptor interface {
 type Proxy interface {
 	Start() error
 	Stop() error
+	// Metrics returns a map of counter-type metrics. Implementations may return zero or more of the metrics defined
+	// below, as well as any number of implementation-defined metrics.
+	Metrics() map[string]uint
 	Force() error
 }
 
+const (
+	// MetricRequests is the total number of requests received by the proxy.
+	MetricRequests = "requests_total"
+	// MetricRequestsExcluded is the total number of requests passed through due to exclusion rules.
+	MetricRequestsExcluded = "requests_excluded"
+	// MetricRequestsFaulted is the total number requests for which the proxy returned an error.
+	MetricRequestsFaulted = "requests_faulted"
```

Comment:
Sorry for not mentioning this earlier, but I think this name is not correct. The most common meaning of "faulted" is "criticized or blamed for something". I suggest "Disrupted" instead.